harness/determined/pytorch/_pytorch_context.py

```diff
@@ -18,6 +18,64 @@
     pass
 
 
+def default_allgather_fn(metrics: Any) -> List:
+    """
+    A noop allgather implementation to ensure that custom reducers work outside of Determined.
+    """
+    return [metrics]
+
+
+class _WrappedReducer:
```

Comment:
I would eventually like it here, but I did move it to _experimental.py for other reasons.

The reason I would like it here is because it is literally only used in this file, and the file may be long but IMO it doesn't seem particularly disorganized or hard to navigate.